docs-search/common/src/Docs/Search/Declarations.purs

```diff
@@ -208,7 +208,7 @@ extractPackageName _ Nothing = UnknownPackage
 extractPackageName _ (Just (SourceSpan { name })) =
   fromMaybe LocalPackage do
     topLevelDir <- dirs !! 0
-    if topLevelDir == ".spago" then Package <<< PackageName <$> dirs !! 1
+    if topLevelDir == ".spago" then Package <<< PackageName <$> dirs !! 2
```

Comment:
We already do a similar matching in the `Purs.Graph` module, and in there we basically:
- construct Globs for each package we have in the build
- [match each module to its glob](https://github.com/purescript/spago/blob/29cd3cad10e0b2dca093ade0a7a2d114186002ba/src/Spago/Purs/Graph.purs#L91)

Not sure if we have a list of packages available here - if we don't we might just want to keep this approach and change how we structure the `packages` folder instead.
The path of a dependency is defined [here](https://github.com/purescript/spago/blob/29cd3cad10e0b2dca093ade0a7a2d114186002ba/src/Spago/Config.purs#L415), so we can change the registry packages to be `packages/$package/$version` rather than `packages/$package-$version`. (if we go for this we'll have to add a `mkdirp` [in Fetch](https://github.com/purescript/spago/blob/29cd3cad10e0b2dca093ade0a7a2d114186002ba/src/Spago/Command/Fetch.purs#L272) to create `packages/$package`)

Reply:
Something else that we should take care of is the `<local package>` bit: the new Spago allows for more than one package in the repo, so we'd need to find a way to split the local files into their proper packages

Reply:
If that makes sense to you then we can just merge this as "good enough" and open an issue to track implementing the "proper way" of getting package info